examples/spineml_braitenberg/spineml_braitenberg.cc

```diff
@@ -0,0 +1,204 @@
+
+        // Add each vertex to the shape
+        for (size_t i = 0; i < original.size(); i++) {
+            m_GreenShape.setPoint(i, renderer.vectorToPixel(original[i]));
+        }
+
+        m_RedShape.setFillColor(sf::Color::Red);
+        for (size_t i = 0; i < original.size(); i++) {
+            m_RedShape.setPoint(i, renderer.vectorToPixel(resized(i, 0), resized(i, 1)));
+        }
+    }
+
+    virtual void draw(sf::RenderTarget &target, sf::RenderStates states) const override
+    {
+        target.draw(m_RedShape, states);
+        target.draw(m_GreenShape, states);
+    }
+
+private:
+    sf::ConvexShape m_GreenShape, m_RedShape;
+};
+
+
+int bob_main(int, char **)
+{
+    // The x and y dimensions of the robot
+    // **THINK** if would be nice if this was part of a robot class?
+    using V = Vector2<meter_t>;
+    constexpr std::array<V, 4> robotDimensions = {
+        V{ -75_mm, 70_mm},
+        V{ 75_mm, 70_mm },
+        V{ 75_mm, -120_mm },
+        V{ -75_mm, -120_mm }
+    };
+
+    // Tank agent
+    Robots::SimulatedTank<> robot(1.0_mps, 104_mm);
+
+    // For displaying the agent
+    SFMLWorld<> display({5_m, 5_m});
+    auto car = display.createCarAgent();
+
+    // Read objects from file
+    const auto objects = readObjects("objects.yaml");
+    CollisionDetector collisionDetector{ robotDimensions, objects, 30_cm, 1_cm };
+
+    // Object size + buffer around
+    const auto &resizedObjects = collisionDetector.getResizedObjects();
+
+    // Create drawable objects
+    std::vector<ArenaObject> objectShapes;
+    objectShapes.reserve(objects.size());
+    for (size_t i = 0; i < objects.size(); i++) {
+        objectShapes.emplace_back(display, objects[i], resizedObjects[i]);
+    }
+
+    // Create SpineML simulator object
+    SpineMLSimulator::Simulator simulator("experiment1.xml", ".");
+
+    // Get external loggers used for accessing steering signals generated by network
+    const auto *steerLeft = simulator.getExternalLogger("steer left");
+    const auto *steerRight = simulator.getExternalLogger("steer right");
+    BOB_ASSERT(steerLeft != nullptr);
+    BOB_ASSERT(steerRight != nullptr);
+    BOB_ASSERT(steerLeft->getModelPropertySize() == 1);
+    BOB_ASSERT(steerRight->getModelPropertySize() == 1);
+
+    // Get external inputs used for providing sensor input to the network
+    auto *sensorLeft = simulator.getExternalInput("sensor left");
+    auto *sensorRight = simulator.getExternalInput("sensor right");
+    BOB_ASSERT(sensorLeft != nullptr);
+    BOB_ASSERT(sensorRight != nullptr);
+
+    while(true) {
+        // Move robot
+        collisionDetector.setRobotPose(robot.getPose());
+
+        // Normalise yaw
+        // **YUCK** why is this necessary? I would expect yaw to remain normalised
+        const auto robotYaw = normaliseAngle180(robot.getPose().yaw());
+
+        // If robot has hit right wall
+        // **TODO** do this more elegantly with err maths
```

Comment:
We could also check for wall collisions by extending the ``CollisionDetector`` class somehow -- I've been meaning to implement something along those lines because it would be handy

Reply:
yeah - that would be a nicer solution - might have a go at that

Reply:
Shall we leave this for a subsequent PR?

Reply:
Yep. Doesn't need doing right now.

Reply:
#121